### PR TITLE
feat: add hilla bom and artifact

### PR DESCRIPTION
**Pull request by haijian-vaadin**
add hilla bom and artifact, so that a user project can depend on hilla instead of vaadin.

### CODE REVIEW

File: hilla/src/main/templates/dev/hilla/shrinkwrap/VaadinCoreShrinkWrap.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * the License.
+ */
+package dev.hilla.shrinkwrap;
+
+import com.vaadin.flow.component.dependency.NpmPackage;
+
+/**
+ * Empty class which makes flow add npm vaadin-core-shrinkwrap dependency.
+ */
+@NpmPackage(value = "@vaadin/vaadin-core-shrinkwrap", version = "${shrink.version}")
```
Comment (manolo) [inline hilla/src/main/templates/dev/hilla/shrinkwrap/VaadinCoreShrinkWrap.java:23]: I think we can add here `@NpmPackage(value="@vaadin/vaadin-core")` and package.json file will be updated with all the components.

Comment (haijian-vaadin) [reply hilla/src/main/templates/dev/hilla/shrinkwrap/VaadinCoreShrinkWrap.java]: Per discussion, removed this file all together

File: hilla/pom.xml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-dev-server</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-client</artifactId>
+        </dependency>
+        <dependency>
```
Comment (manolo) [inline hilla/pom.xml:49]: We must remove lumo and material dependency here.
A pure TS project might want not to use vaadin-components nor vaadin-themes 

But previously we need to fix https://github.com/vaadin/flow/issues/12606

**Review by manolo — CHANGES_REQUESTED** [review]

**Comment by manolo** [comment]
Before merging this, we should fix https://github.com/vaadin/flow/issues/12606

File: hilla/pom.xml (hunk 2 above)
Comment (haijian-vaadin) [reply hilla/pom.xml:49]: hmm, I think it's fine to have those dependencies for now. As the main goal is to have a `hilla-bom` and dependency that a user can use.
Separate Hilla from Flow is the ultimate goal, but not necessarily to have it now. For example, `fusion-endpoint` depends on `flow-server` anyway.

**Review by manolo — APPROVED** [review]

File: hilla/pom.xml (hunk 2 above)
Comment (manolo) [reply hilla/pom.xml:49]: ok, let lave it for now

**Review by ZheSun88 — APPROVED** [review]

**Pull request merged**